File: server/models/Snippet.js
const Sequelize = require("sequelize");
const db = require("../utils/database");

module.exports = db.define("snippets", {
    name: {
        type: Sequelize.STRING,
        allowNull: false,
    },
    accountId: {
        type: Sequelize.INTEGER,
        allowNull: true,
        references: { model: "accounts", key: "id" },
        onDelete: "CASCADE",
    },
    organizationId: {
        type: Sequelize.INTEGER,
        allowNull: true,
        references: { model: "organizations", key: "id" },
        onDelete: "CASCADE",
    },
    sourceId: {
        type: Sequelize.INTEGER,
        allowNull: true,
        references: { model: "sources", key: "id" },
        onDelete: "CASCADE",
    },
    command: {
        type: Sequelize.TEXT,
        allowNull: false,
    },
    description: {
        type: Sequelize.TEXT,
        allowNull: true,
    },
    sortOrder: {
        type: Sequelize.INTEGER,
        allowNull: false,
        defaultValue: 0,
    },
    osFilter: {
        type: Sequelize.JSON,
        allowNull: true,
        defaultValue: null,
    }
}, {
    freezeTableName: true,
    createdAt: false,
    updatedAt: false,
    hooks: {
        afterFind: (results) => {
            const parse = (item) => {
                if (item?.osFilter && typeof item.osFilter === 'string') {
                    try { item.osFilter = JSON.parse(item.osFilter); } catch {}
                }
            };
            Array.isArray(results) ? results.forEach(parse) : parse(results);
        },
    },
});

File: mobile/lib/models/snippet.dart
class Snippet {
  final int id;
  final String name;
  final int accountId;
  final String command;
  final String? description;

  const Snippet({required this.id, required this.name, required this.accountId, required this.command, this.description});

  factory Snippet.fromJson(Map<String, dynamic> json) => Snippet(
    id: json['id'] as int,
    name: json['name'] as String? ?? '',
    accountId: json['accountId'] as int,
    command: json['command'] as String? ?? '',
    description: json['description'] as String?,
  );
}
